Google.PowerShell/Common/AuthenticateWithSdkCredentialsExecutor.cs

```diff
@@ -36,13 +36,14 @@ public class AuthenticateWithSdkCredentialsExecutor : ICredential, IHttpExecuteI
         /// rather than launching gcloud again.
         /// </summary>
         private static ActiveUserToken s_token;
-        private static object s_lockObject = new Object();
+        private static readonly SemaphoreSlim s_tokenLock = new SemaphoreSlim(1);
 
         /// <summary>
-        /// We just need a default flow for things like flow.AccessMethod.Intercept
-        /// or flow.RevokeTokenAsync.
+        /// Returns the authorization code flow. 
+        /// This is used to revoke token (in RevokeTokenAsync)
+        /// and intercept request with the access token (in InterceptAsync).
         /// </summary>
-        private readonly IAuthorizationCodeFlow flow = new GoogleAuthorizationCodeFlow(
+        public IAuthorizationCodeFlow Flow { get; } = new GoogleAuthorizationCodeFlow(
```

Comment:
Given that `Flow` is only used inside of this class, perhaps makes sense tom make it private?